brainglobe_stitch/image_mosaic.py

```diff
@@ -340,3 +346,366 @@ def read_big_stitcher_transforms(self) -> None:
+        compression: str = "zstd",
+        compression_level: int = 3,
+        downscale_factors: Tuple[int, int, int] = (1, 2, 2),
+    ) -> None:
+        """
+        Fuse the tiles in the ImageMosaic into a single image and save it as a
+        zarr file.
+
+        Parameters
+        ----------
+        output_path: Path
+            The path of the output file.
+        fused_image_shape: Tuple[int, ...]
+            The shape of the fused image.
+        pyramid_depth: int, default: 5
+            The depth of the resolution pyramid.
+        chunk_shape: Tuple[int, ...], default: (128, 128, 128)
+            The shape of the chunks in the zarr file.
+        compression: str, default: "zstd"
+            The compression algorithm to use.
+        compression_level: int, default: 3
+            The compression level to use.
+        downscale_factors: Tuple[int, int, int], default: (1, 2, 2)
+            The factors to downscale the image by in the z, y, x dimensions.
+        """
+        z_size, y_size, x_size = self.tiles[0].data_pyramid[0].shape
+
+        transformation_metadata, axes_metadata = self.get_metadata_for_zarr(
+            pyramid_depth=pyramid_depth, downscale_factors=downscale_factors
+        )
+
+        if self.num_channels > 1:
+            fused_image_shape = (self.num_channels, *fused_image_shape)
+            chunk_shape = (self.num_channels, *chunk_shape)
+
+        store = zarr.NestedDirectoryStore(str(output_path))
+        root = zarr.group(store=store)
+        compressor = Blosc(
+            cname=compression, clevel=compression_level, shuffle=Blosc.SHUFFLE
+        )
+
+        fused_image_store = root.create(
+            "0",
+            shape=fused_image_shape,
+            chunks=chunk_shape,
+            dtype="i2",
+            compressor=compressor,
+        )
+
+        # Place the tiles in reverse order of acquisition
+        for tile in self.tiles[-1::-1]:
+            if self.num_channels > 1:
+                fused_image_store[
+                    tile.channel_id,
+                    tile.position[0] : tile.position[0] + z_size,
+                    tile.position[1] : tile.position[1] + y_size,
+                    tile.position[2] : tile.position[2] + x_size,
+                ] = tile.data_pyramid[0].compute()
+            else:
+                fused_image_store[
+                    tile.position[0] : tile.position[0] + z_size,
+                    tile.position[1] : tile.position[1] + y_size,
+                    tile.position[2] : tile.position[2] + x_size,
+                ] = tile.data_pyramid[0].compute()
+
+            print(f"Done tile {tile.id}")
+
+        for i in range(1, len(transformation_metadata)):
+            prev_resolution = da.from_zarr(
+                root[str(i - 1)], chunks=chunk_shape
+            )
+
+            if self.num_channels > 1:
+                downsampled_image = downscale_nearest(
+                    prev_resolution, (1, *downscale_factors)
```

Comment:
maybe `factors = (1, *downscale_factors) if self.num_channels > 1 else downscale_factors)` then a single call to `downscale_nearest` to reduce duplication?